Assets/Mirror/Runtime/NetworkServer.cs

```diff
@@ -970,18 +970,24 @@ internal static void SendSpawnMessage(NetworkIdentity identity, NetworkConnectio
             }
         }
 
+        // destroy the player and all their possessions
         public static void DestroyPlayerForConnection(NetworkConnection conn)
         {
-            // => destroy what we can destroy.
-            HashSet<uint> tmp = new HashSet<uint>(conn.clientOwnedObjects);
-            foreach (uint netId in tmp)
+            foreach (uint netId in conn.clientOwnedObjects)
             {
                 if (NetworkIdentity.spawned.TryGetValue(netId, out NetworkIdentity identity))
                 {
+                    // note: clear the owner before destroying, because DestroyObject
+                    // tries to remove this object from the authority owner and fails
+                    // because we are foreach'ing over that collection.
+                    identity.clientAuthorityOwner = null;
```

Comment:
that's a bit strange imho.
we making assumptions about what DestroyObject does internally and try to do this here already.
this is a bit too much magic. isn't there a better solution?